std/typecons.d

```diff
@@ -64,67 +69,44 @@ static if (is(T:Object))
 else
     alias RefT = T*;
```

Comment:
We should probably choose `scoped!T` as storage for classes, what does RefCounted do here?